commit 7569f1e2f942eeba02c918454980e4cd7741042a
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Thu Feb 26 09:58:22 2015 +0000

    pinctrl-bcm2835: Only request the interrupts listed in the DTB
    
    Although the GPIO controller can generate three interrupts (four counting
    the common one), the device tree files currently only specify two. In the
    absence of the third, simply don't register that interrupt (as opposed to
    registering 0), which has the effect of making it impossible to generate
    interrupts for GPIOs 46-53 which, since they share pins with the SD card
    interface, is unlikely to be a problem.

diff --git a/drivers/pinctrl/bcm/pinctrl-bcm2835.c b/drivers/pinctrl/bcm/pinctrl-bcm2835.c
index 5b7cb4c415e1..6351fe7f8e31 100644
--- a/drivers/pinctrl/bcm/pinctrl-bcm2835.c
+++ b/drivers/pinctrl/bcm/pinctrl-bcm2835.c
@@ -1049,6 +1049,8 @@ static int bcm2835_pinctrl_probe(struct platform_device *pdev)
 		int len;
 		char *name;
 		pc->irq[i] = irq_of_parse_and_map(np, i);
+		if (pc->irq[i] == 0)
+			break;
 		pc->irq_data[i].pc = pc;
 		pc->irq_data[i].irqgroup = i;
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1030_7569f1e2f942eeba02c918454980e4cd7741042a.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1030_7569f1e2f942eeba02c918454980e4cd7741042a.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
